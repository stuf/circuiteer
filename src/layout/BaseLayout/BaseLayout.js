import { Header } from './components';

export function BaseLayout(props) {
  const { children } = props;

  return (
    <main
      className={`
    h-full
    flex flex-col
    `}
    >
      <Header
        className={`
         h-12
      `}
      >
        <div className="space-x-1">
          <button
            className={`
          bg-purple-500
          text-white
          px-2 py-1
          rounded-lg
          `}
          >
            Thing 1
          </button>
        </div>
      </Header>

      {/* Main content area */}
      <div
        className={`
        flex-grow
        overflow-auto
      `}
      >
        {children}
      </div>
    </main>
  );
}
