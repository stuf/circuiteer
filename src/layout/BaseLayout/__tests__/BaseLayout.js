import { render } from '@testing-library/react';
import { BaseLayout } from '../BaseLayout';

describe('BaseLayout', () => {
  test('works', () => {
    const { container } = render(<BaseLayout />);

    expect(container).toMatchSnapshot();
  });
});
