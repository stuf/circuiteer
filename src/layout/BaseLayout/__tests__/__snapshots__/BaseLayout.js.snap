// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout works 1`] = `
<div>
  <main
    class="
    h-full
    flex flex-col
    "
  >
    <header
      class="base-layout__header 
         h-12
      "
    >
      <div
        class="flex items-center space-x-2 w-full px-4 py-2"
      >
        <div
          class="space-x-2"
        >
          <span
            class="font-bold"
          >
            Circuiteer
          </span>
          <sup
            class="text-gray-500"
          >
            Experimental
          </sup>
        </div>
        <div
          class="flex-grow"
        />
        <div
          class=""
        />
      </div>
    </header>
    <div
      class="
        flex-grow
        overflow-auto
      "
    />
  </main>
</div>
`;
